sources/core/Stride.Core.DiagnosticsServices/Diagnoser.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Microsoft.CodeAnalysis;
+using StrideDiagnostics.PropertyFinders;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace StrideDiagnostics;
+
+internal class Diagnoser
+{
+    internal void StartCreation(ClassInfo info)
+    {
+        DiagnoseDataMember(info);
+    }
+    private void DiagnoseDataMember(ClassInfo info)
+    {
+        IEnumerable<Type> reporterTypes = typeof(Diagnoser).Assembly.GetTypes()
```

Comment:
This could be cached by assigning it to a static field
```csharp
private static List<Type> reporterTypes = typeof(Diagnoser).Assembly.GetTypes()
    .Where(type => typeof(IViolationReporter).IsAssignableFrom(type) && !type.IsInterface && !type.IsAbstract)
    .ToList();
```

Reply:
c0592db

is applied